src/main/java/org/octri/hpoonfhir/view/PhenotypeModel.java

```diff
@@ -0,0 +1,57 @@
+package org.octri.hpoonfhir.view;
+
+import java.io.Serializable;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.monarchinitiative.fhir2hpo.hpo.HpoTermWithNegation;
+
+/**
+ * The model representing the fields on the phenotype that will be displayed.
+ * 
+ * @author yateam
+ *
+ */
+public class PhenotypeModel implements Serializable {
+	
+	private static final long serialVersionUID = 5911092102035916719L;
+
+	private final String hpoTermName;
+	private final String hpoTermId;
+	private final String first;
+	private final String last;
+	private final List<ObservationModel> observations;
+	
+	public PhenotypeModel(HpoTermWithNegation hpoTerm, List<ObservationModel> observations) {
+		this.hpoTermName = hpoTerm.toString();
+		this.hpoTermId = hpoTerm.getHpoTerm().getId().getIdWithPrefix();
+		this.observations = observations;
+		this.first = observations.stream().filter(observation -> !StringUtils.isBlank(observation.getDate())).reduce((x, y) -> x.getDate().compareTo(y.getDate()) < 0 ? x:y).get().getDate();
+		this.last = observations.stream().filter(observation -> !StringUtils.isBlank(observation.getDate())).reduce((x, y) -> x.getDate().compareTo(y.getDate()) < 0 ? y:x).get().getDate();
```

Comment:
I think these could be simplified to something like:

  // you might still need the filter
   minDate = observations.stream().map(obs -> obs.getDate()).min(String::compareTo).get();